src/module/system/text-editor.ts

```diff
@@ -228,6 +229,77 @@ class TextEditorPF2e extends TextEditor {
         return roll.toMessage({ speaker, flavor: roll.options.flavor }, { rollMode });
     }
 
+    static override async _applyCustomEnrichers(
+        pattern: RegExp,
+        enricher: EnricherFunction,
+        text: Text[],
+        options: EnrichmentOptionsPF2e = {},
+    ): Promise<boolean> {
+        // Always apply normal enrichers first
+        const result = await superApplyCustomEnrichers.apply(this, [pattern, enricher, text, options]);
+        // Replace any damage rolls
+        const damageResult = await TextEditorPF2e.#replaceDamageRolls(text, options);
+        return damageResult || result;
+    }
+
+    static async #replaceDamageRolls(text: Text[], options: EnrichmentOptionsPF2e): Promise<boolean> {
+        const getLastBracketIndex = (formula: string, firstBracketIndex: number): number | null => {
+            if (formula.length === 0) return null;
+            const brackets = { open: 0, lastIndex: 0 };
+            for (const [index, c] of Array.from(formula).entries()) {
+                if (c === "[") brackets.open += 1;
+                if (c === "]") {
+                    brackets.open -= 1;
+                    brackets.lastIndex = index;
+                    if (brackets.open === 0) break;
+                }
+            }
+            return brackets.open === 0 && brackets.lastIndex > 0 ? firstBracketIndex + brackets.lastIndex : null;
+        };
+        const data: { replaced: boolean; target: Text | null } = { replaced: false, target: null };
```

Comment:
I don't think I'm a fan of this. The data.blah actually threw me off a bit until I realized you were trying to do let without using let. I think using let here is fine and even preferred.